### PR TITLE
Fix bugs with renaming rooms

### PR DESCRIPTION
**Pull request by whalemer**
Recreate the logs properly, fix aliases

Incidentally, should the rename command modlog, or log to the staff room?

### CODE REVIEW

File: server/rooms.ts [hunk 1]
```diff
@@ -1368,6 +1368,12 @@ export class BasicChatRoom extends BasicRoom {
 		Rooms.rooms.delete(oldID);
 		Rooms.rooms.set(newID, this as ChatRoom);
 
+		for (const [alias, roomid] of Rooms.aliases.entries()) {
+			if (roomid === oldID) {
+				Rooms.aliases.set(alias, newID);
```
Comment (TheJetOU) [inline server/rooms.ts]: You should delete the old aliases now that it's useless.

Comment (whalemer) [reply server/rooms.ts]: It's an alias -> roomid map, I'm just updating it from `alias -> oldid` to `alias -> newid`

Comment (TheJetOU) [reply server/rooms.ts]: Ah, right.

Comment (Zarel) [inline server/rooms.ts]: Unrelatedly, do we set an alias from the old room to the new room? That should probably be done.

**Comment by Zarel** [comment]
Notifying Staff is weird because secret rooms usually only notify Upper Staff. I'm fine with the exception but we might want to think about something more consistent.

**Comment by whalemer** [comment]
>Notifying Staff is weird because secret rooms usually only notify Upper Staff. 

Isn't that what my code does? see https://github.com/whalemer/Pokemon-Showdown/blob/c3cf7d4e721d77a5da3630eb6d1caa1972b5d2cc/server/chat-commands/room-settings.ts#L833-L834

File: server/rooms.ts (hunk 1 above)
Comment (whalemer) [reply server/rooms.ts]: Sure.

**Comment by Zarel** [comment]
Ah, I missed that while reading.

**Pull request merged**